wave_calculator.gd

```diff
@@ -6,19 +6,50 @@ var rng: RandomNumberGenerator = RandomNumberGenerator.new()
 var _base_spawn_time: float
 var _enemies: Dictionary  # Was not able to type the dict either
 
+const MAX_WAVE_BATCH_SCORE = 10
+
 
 func _init(base_spawn_time: float, enemies: Dictionary) -> void:
 	_base_spawn_time = base_spawn_time
 	_enemies = enemies
 
 
 # Array[Dictionary] doesn't work :sad: https://forum.godotengine.org/t/array-with-dictionaries-doesnt-get-recognized-as-array-dictionary/58280
-func get_enemy_spawns(wave_number: int) -> Array:
```

Comment:
will see if i can have better typing here - in gdscript `Array[Array[int]]` is not supported